### PR TITLE
Widget showcase tutorial

### PR DESCRIPTION
**Pull request by BaoTon**
Add the second tutorial demonstrating how to bring widgets from the showcase to the iTwin viewer.

### CODE REVIEW

File: docs/learning/tutorials/showcase-to-viewer.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+### Understanding the Sample Showcase
+
+We hope you have given the [sample showcase](https://www.itwinjs.org/sample-showcase) a tour and enjoyed the many samples we provide. You may want to use some of these samples in your own iTwin Viewer, and to do so you'll first need to understand how the showcase works.
+
+If we take a closer look at the files involved in each sample, you'll notice they all follow the same pattern containing a few important files:
+
+1. ```[SampleName]App.tsx``` - Corresponds to ```App.tsx``` in the itwin viewer template and provides the main ```Viewer``` component.
+2. ```[SampleName]Widget.tsx``` - Defines the ```UiItemsProvider``` that will be passed into prop ```uiProviders``` for our sample widget component. This widget is the controller for our samples.
+3. ```[SampleName]Api.ts``` - Defines widget functionality that uses the itwin.js API being showcased.
```
Comment (calebmshafer) [inline docs/learning/tutorials/showcase-to-viewer.md:26]: itwin.js -> iTwin.js

File: docs/learning/tutorials/showcase-to-viewer.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+This tutorial will take widgets from the sample showcase and add them into your iTwin Viewer using the ```uiProviders``` prop.
+
+### Understanding the Sample Showcase
+
+We hope you have given the [sample showcase](https://www.itwinjs.org/sample-showcase) a tour and enjoyed the many samples we provide. You may want to use some of these samples in your own iTwin Viewer, and to do so you'll first need to understand how the showcase works.
+
+If we take a closer look at the files involved in each sample, you'll notice they all follow the same pattern containing a few important files:
+
+1. ```[SampleName]App.tsx``` - Corresponds to ```App.tsx``` in the itwin viewer template and provides the main ```Viewer``` component.
```
Comment (calebmshafer) [inline docs/learning/tutorials/showcase-to-viewer.md:24]: itwin viewer -> iTwin Viewer

**Review by calebmshafer — CHANGES_REQUESTED** [review]
Is there an entry point to this tutorial already?

File: docs/learning/tutorials/showcase-to-viewer.md (hunk 1 above)
Comment (BaoTon) [reply docs/learning/tutorials/showcase-to-viewer.md]: Added a new tile in the tutorials section for the entry.

**Comment by BaoTon** [comment]
@Mergifyio backport release/2.15.x

File: docs/learning/tutorials/showcase-to-viewer.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+Given this pattern, it's simple to identify the parts required to bring our sample showcase to your iTwin Viewer. The component revolves around the ```[SampleName]Widget.tsx``` file so we need to copy all the files associated with our custom widget's ```UiItemsProvider``` and pass it in the ```Viewer``` component.
+
+### Example using a sample
+
+For this tutorial, we'll be taking the widget from sample [View Attributes](https://www.itwinjs.org/sample-showcase/?group=Viewer&sample=view-attributes-sample&imodel=House+Sample) and adding it into our iTwin Viewer.
+
+Starting with our entry file ```ViewAttributesApp.tsx```, the ```Viewer``` component is defined as follows:
+
+``` HTML
+        <Viewer
```
Comment (williamkbentley) [inline docs/learning/tutorials/showcase-to-viewer.md:38]: Weird tabbing:
image

**Review by williamkbentley — CHANGES_REQUESTED** [review]

File: docs/learning/tutorials/showcase-to-viewer.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
+# Adding showcase widgets to your iTwin Viewer
+
+### Prerequisites
+
+This tutorial assumes that you already have:
+
+- Explored the [sample showcase](https://www.itwinjs.org/sample-showcase).
+- Your own local source for the iTwin Web Viewer based on the template @bentley/itwin-viewer
```
Comment (williamkbentley) [inline docs/learning/tutorials/showcase-to-viewer.md:8]: Could add a link to itwin-viewer repo

File: docs/learning/tutorials/showcase-to-viewer.md (hunk 3 above)
Comment (BaoTon) [reply docs/learning/tutorials/showcase-to-viewer.md]: For some reason, it doesn't preserve my tabbing. I left-justified it instead.

File: docs/learning/tutorials/showcase-to-viewer.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+To sum it up, these three lines are the only lines you'll need to add in your iTwin Viewer in ```App.tsx```:
+
+``` typescript
+... // Import the widget provider
+import { ViewAttributesWidgetProvider } from "./ViewAttributesWidget";
+... // Construct the widget provider
+const uiProviders = [new ViewAttributesWidgetProvider()];
+... // Passed into the viewer component
```
[the remark this answers is not in the document]
Comment (BaoTon) [reply docs/learning/tutorials/showcase-to-viewer.md]: Specified 'Prop passed into the <Viewer> component'

File: docs/learning/tutorials/showcase-to-viewer.md (hunk 4 above)
Comment (BaoTon) [reply docs/learning/tutorials/showcase-to-viewer.md]: Added link to itwin-viewer repo on 'local source'

**Review by williamkbentley — APPROVED** [review]

**Review by calebmshafer — CHANGES_REQUESTED** [review]

File: docs/learning/tutorials/showcase-to-viewer.md [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+For this tutorial, we'll be taking the widget from sample [View Attributes](https://www.itwinjs.org/sample-showcase/?group=Viewer&sample=view-attributes-sample&imodel=House+Sample) and adding it into our iTwin Viewer.
+
+Starting with our entry file ```ViewAttributesApp.tsx```, the ```Viewer``` component is defined as follows:
+
+``` jsx
+<Viewer
+  contextId={sampleIModelInfo.contextId}
+  iModelId={sampleIModelInfo.iModelId}
+  authConfig={{ oidcClient: AuthorizationClient.oidcClient }}
+  viewportOptions={viewportOptions}
```
[the remark this answers is not in the document]
Comment (BaoTon) [reply docs/learning/tutorials/showcase-to-viewer.md:42]: The other props aren't required, I added a brief description that they are used specifically for showcase setup.

[the remark this answers is not in the document]
Comment (calebmshafer) [reply docs/learning/tutorials/showcase-to-viewer.md:42]: Did you push your changes?  I don't see them.

**Review by calebmshafer — APPROVED** [review]

**Comment by BaoTon** [comment]
@Mergifyio backport release/2.16.x

**Pull request merged**